### ISSUE TITLE
Add support for building with system rapidfuzz-cpp

### ISSUE THREAD

**Issue by SuperSandro2000** [maxbachmann/Levenshtein#22]
It would be nice for distros if building with a system rapidfuzz was supported.

**Comment by maxbachmann** [comment]
I guess this would be more of a request for [RapidFuzz](https://github.com/maxbachmann/RapidFuzz). I am however unsure whats the use case for this. Could you explain, whats the advantage of building against a system supplied version of `rapidfuzz-cpp` instead of building against the version included in the sdist?

**Comment by SuperSandro2000** [comment]
I just updated the rapidfuzz/levenshtein/rapidfuzz-capi package in NixOS, a Linux Distro, and we are trying to avoid cloning repositories with submodules to make sure we use the same version of the package everywhere. This makes security updates and patching in general easier and makes sure we are using the same version of a package across all python packages.

**Comment by maxbachmann** [comment]
Is there any reason why you can not build from the sdist on Pypi? This is generally the recommended source for package maintainers. I require e.g. Cython3.0 when building from git, while the sdist already includes the generated C++ files. This makes building a lot simpler. I do this when building for conda forge.

**Comment by SuperSandro2000** [comment]
Its not about the submodule in itself. We can clone that, too but more about the vendored source code. So it is the same problem with the sdist. It also does not contain the tests which we always want to run to make sure things work like they should especially if we take version constraints a bit light hearted.

Cython 3 is not such a problem. Normally we try to avoid multiple versions of package as much as possible because python itself can't handle multiple versions of the same package in the same environment which leads to strange and hard debug errors but since Cython is only used at build time that is not a problem.

**Comment by maxbachmann** [comment]
The build script is CMake based, so it is no problem to change it in a way, that uses a rapidfuzz version installed on the system. 
However I am unsure how you would select, that you would like to use the system version of the libraries. How would you be able to pass this setting into the setup.py.

**Comment by maxbachmann** [comment]
I guess I could just usw the system lib if available + in the correct version and use FetchContent as fallback.

**Comment by maxbachmann** [comment]
https://github.com/maxbachmann/RapidFuzz/pull/213 is a work in progress, which uses system provided versions of `rapidfuzz-cpp` and `jarowinkler-cpp` if available.

**Comment by SuperSandro2000** [comment]
> The build script is CMake based, so it is no problem to change it in a way, that uses a rapidfuzz version installed on the system. However I am unsure how you would select, that you would like to use the system version of the libraries. How would you be able to pass this setting into the setup.py.

I have no experience with using cmake in setup.py.

> I guess I could just usw the system lib if available + in the correct version and use FetchContent as fallback.

That could work.

**Comment by maxbachmann** [comment]
Can you try this with Rapidfuzz v2.0.10? It should now use system installed versions of Taskflow / rapidfuzz-cpp / jarowinkler-cpp as long as they are in the correct version:
Taskflow: 2.7.0
rapidfuzz-cpp: 1.0.1
jarowinkler-cpp: 1.0.0

When building in verbose mode it should either print:
```
Using system supplied version of ...
```
or
```
Using FetchContent to load ...
```
depending on the version it is using.

**Comment by maxbachmann** [comment]
> I can confirm that rapidfuzz 2.0.10 builds with system libraries if they are available.

:+1:

> Is there any reason to use an old version of Taskflow?

Some platforms still come with compilers which only support C++14. For this reason I am still supporting C++14. Taskflow 2.7.0 is the latest version of Taskflow which supports C++14. However it would probably be reasonable to accept newer versions when they are installed, since it does not make much sense for someone to install Taskflow 3.3.0 on a platform without a C++17 compiler ...

> I don't think it's necessary to require an exact version for the libraries

Yes I could relax those requirements considering all three libraries specify `COMPATIBILITY SameMajorVersion`

**Comment by dotlambda** [comment]
While RapidFuzz now allows using a system-supplied rapidfuzz-cpp, Levenshtein doesn't yet.

**Comment by maxbachmann** [comment]
Ah yes forgot this change is required here as well. I can make a release for the `Levenshtein` library with the change tomorrow

**Comment by maxbachmann** [comment]
This should be fixed in `v0.18.2`

**Comment by dotlambda** [comment]
> This should be fixed in `v0.18.2`

It is: https://github.com/NixOS/nixpkgs/pull/180250
Thank you!

**Comment by maxbachmann** [comment]
For Jarowinkler/Rapidfuzz you added the changelog. The changelog for the Levenshtein package can be found here: https://github.com/maxbachmann/Levenshtein/blob/main/HISTORY.md
